blog/2017-02-03-using-native-driver-for-animated.md

```diff
@@ -0,0 +1,146 @@
+```
+NativeAnimatedModule.startAnimation({
+  type: 'timing',
+  frames: [0, 0.1, 0.2, 0.4, 0.65, ...],
+  animatedValueId: 1,
+});
+```
+
+And now here's the breakdown of what happens when the animation runs:
+- Native: The native animation driver uses `CADisplayLink` or `android.view.Choreographer` to execute on every frame and update the value it drives using the new value it calculates based on the animation curve.
+- Native: Intermediate values are calculated and passed to a props node that is attached to a native view.
+- Native: The `UIView` or `android.View` is updated.
+
+As you can see no more JS thread and no more bridge which means faster animations! 🎉🎉
+
+## How do I use this in my app?
+
+For normal animations the answer is pretty simple, you just have to add `useNativeDriver: true` to the animation config when starting it.
+
+Before:
+```
+Animated.timing(this.state.animatedValue, {
+  toValue: 1,
+  duration: 500,
+});
+```
+After:
+```
+Animated.timing(this.state.animatedValue, {
+  toValue: 1,
+  duration: 500,
+  useNativeDriver: true, // <-- Add this
+}).start();
+```
+
+Animated values are only compatible with one driver so if you use native driver when starting an animation on a value, make sure every animations on that value also use the native driver.
+
+It also works with `Animated.event`, this is very useful if you have an animation that must follow the scroll position because without the native driver it will always run a frame behind of the gesture because of the async nature of React Native.
+
+Before:
+```
+<ScrollView
+  scrollEventThrottle={16}
+  onScroll={Animated.event(
+    [{ nativeEvent: { contentOffset: { y: this.state.animatedValue } } }]
+  )}
+>
+  {content}
+</ScrollView>
+```
+After:
+```
+<Animated.ScrollView // <-- Use the Animated ScrollView wrapper
+  scrollEventThrottle={1} // <-- Use 1 here to make sure no events are ever missed
+  onScroll={Animated.event(
+    [{ nativeEvent: { contentOffset: { y: this.state.animatedValue } } }],
+    { useNativeDriver: true } // <-- Add this
+  )}
+>
+  {content}
+</Animated.ScrollView>
+```
+
+## Caveats
+
+Not everything you can do with Animated is currently supported in Native Animated. The main limitation is around `Animated.event`, it will only work with direct events and not bubbling events. This means it does not work with `PanResponder` but does work with things like `ScrollView#onScroll`.
+
+Native Animated has also been part of React Native for quite a while but has never been documented because it was considered experimental. Because of that make sure you are using a recent version (0.40+) of React Native if you want to use this feature.
+
+## Resources
+
+I really recommend this talk by [Krzysztof Magiera](https://twitter.com/kzzzf) about animations in React Native: https://www.youtube.com/watch?v=qgSMjYWqBk4.
```

Comment:
perhaps the youtube link I commented above for the talk by vjeux is better suited for here actually